src/hooks/vitest.ts

```diff
@@ -0,0 +1,197 @@
+import assert from "node:assert";
+import { pathToFileURL } from "node:url";
+import worker from "node:worker_threads";
+
+import { simple as walk } from "acorn-walk";
+import { type ESTree } from "meriyah";
+
+import { args as args_, assignment, call_, identifier, literal, member, ret } from "../generate";
+import { info, warn } from "../message";
+import { recording, start } from "../recorder";
+import Recording from "../Recording";
+import genericTransform from "../transform";
+
+function createInitChannel() {
+  return new worker.BroadcastChannel("appmap-node/vitest/initialized");
+}
```

Comment:
Why this function? Seems like you should be able to just do `const initChannel = new worker.BroadcastChannel(...)` on the toplevel. Or will it lead to hanging (which is kinda weird, but if so, ok)?

Reply:
Yes, exactly, it should be called only from the right threads.